src/mame/drivers/h01x.cpp

```diff
@@ -10,6 +13,39 @@
 static const int16_t speaker_levels[] = {-32768, 0, 32767, 0};
 //static int16_t speaker_levels[] = { 0, -32767, 0, 32767 };
 
+void h01x_state::h01b(machine_config &config)
+{
+	// basic machine hardware
+	Z80(config, m_maincpu, 10.6445_MHz_XTAL / 6);
+	m_maincpu->set_addrmap(AS_PROGRAM, &h01x_state::h01x_mem_map);
+	m_maincpu->set_addrmap(AS_IO, &h01x_state::h01x_io_map);
+
+	// RAM
+	RAM(config, m_ram).set_default_size("32K").set_default_value(0x00);
+	// VRAM 16K 4bit
+	RAM(config, m_vram).set_default_size("16K").set_default_value(0xf0);
+
+	// video hardware
+	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
+	screen.set_raw(10.6445_MHz_XTAL, 336, 0, 336, 192, 0, 192);
+	screen.set_screen_update(FUNC(h01x_state::screen_update_h01x));
+	screen.set_palette("palette");
+
+	PALETTE(config, "palette", palette_device::MONOCHROME);
+
+	// sound hardware
+	SPEAKER(config, "mono").front_center();
+	SPEAKER_SOUND(config, m_speaker).set_levels(4, speaker_levels);
+	m_speaker->add_route(ALL_OUTPUTS, "mono", 0.50);
+
+	/* devices */
+	CASSETTE(config, m_cassette);
+	m_cassette->add_route(ALL_OUTPUTS, "mono", 0.05);
+	m_cassette->set_formats(trs80l2_cassette_formats);
+	m_cassette->set_default_state(CASSETTE_STOPPED);
+	//m_cassette->set_default_state(CASSETTE_PLAY);
+}
```

Comment:
Can you please change this so the common parts of the `h01b` and `nf500a` machine configurations are in a common function, and the specific machine configuration functions only add the parts that are different?

Reply:
Thanks for your advice. I tried to change it.